contracts/minter/src/contract.rs

```diff
@@ -0,0 +1,602 @@
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{
+    coin, to_binary, Addr, BankMsg, Binary, Coin, CosmosMsg, Decimal, Deps, DepsMut, Empty, Env,
+    MessageInfo, Order, Reply, ReplyOn, StdError, StdResult, Timestamp, Uint128, WasmMsg,
+};
+use cw2::set_contract_version;
+use cw721::TokensResponse as Cw721TokensResponse;
+use cw721_base::{msg::ExecuteMsg as Cw721ExecuteMsg, MintMsg};
+use cw_utils::{may_pay, parse_reply_instantiate_data, Expiration};
+use sg721::msg::{InstantiateMsg as Sg721InstantiateMsg, QueryMsg as Sg721QueryMsg};
+use url::Url;
+
+use crate::error::ContractError;
+use crate::msg::{
+    ConfigResponse, ExecuteMsg, InstantiateMsg, MintPriceResponse, MintableNumTokensResponse,
+    QueryMsg, StartTimeResponse,
+};
+use crate::state::{Config, CONFIG, MINTABLE_TOKEN_IDS, SG721_ADDRESS};
+use sg_std::{burn_and_distribute_fee, StargazeMsgWrapper, GENESIS_MINT_START_TIME, NATIVE_DENOM};
+use whitelist::msg::{
+    ConfigResponse as WhitelistConfigResponse, HasMemberResponse, QueryMsg as WhitelistQueryMsg,
+    UnitPriceResponse,
+};
+
+pub type Response = cosmwasm_std::Response<StargazeMsgWrapper>;
```

Comment:
This could be used by many smart contracts. Maybe we can pull this to a package?

Reply:
Good idea, but it depends on a custom SDK module.